transformers/tyntecWhatsAppTransformer/transformer-v4.ts

```diff
@@ -0,0 +1,473 @@
+	type: 'button';
+	subType: 'quick_reply';
+	index: number;
+	parameters: IWhatsAppTemplateButtonComponentParameter[];
+}
+
+interface IWhatsAppTemplateButtonComponentParameter {
+	type: 'payload';
+	payload: string;
+}
+
+interface IWhatsAppTemplateComponentTextParameter {
+	type: 'text';
+	text: string;
+}
+
+type TWhatsAppContent = IWhatsAppTextMessage | IWhatsAppMediaMessage | IWhatsAppTemplateMessage | IWhatsAppLocationMessage;
+
+/**
+ * Webchat Interface
+ */
+
+interface IDefaultQuickReply {
+	title: string;
+	payload: string;
+}
+
+interface ISessionStorageQuickReply {
+	index: number;
+	quickReply: IDefaultQuickReply;
+}
+
+const createWhatsAppQuickReplies = (quickReplies: IDefaultQuickReply[], sessionStorage: any): string => {
+
+	// get previous quick replies from session storage 
+	let sessionquickReplyCurrentNumber: number = sessionStorage.quickReplyCurrentNumber || 0;
+	let sessionQuickReplies: ISessionStorageQuickReply[] = sessionStorage.quickReplies || [];
+
+	// initialize empty text message bubble
+	let whatsAppQuickReplyMessage: string = "";
+
+	for (let quickReply of quickReplies) {
+		// store the index to the session storage for further quick replies
+		sessionquickReplyCurrentNumber += 1;
+		sessionQuickReplies.push({
+			index: sessionquickReplyCurrentNumber,
+			quickReply
+		})
+		// add the quick reply to the text message bubble
+		// Example: 1. first quick reply
+		whatsAppQuickReplyMessage += `\n${sessionquickReplyCurrentNumber}. ${quickReply.title}`;
+
+	}
+
+	sessionStorage.quickReplyCurrentNumber = sessionquickReplyCurrentNumber;
+	sessionStorage.quickReplies = sessionQuickReplies;
+
+	return whatsAppQuickReplyMessage;
+}
+
+const convertWebchatContentToWhatsApp = (processedOutput, sessionId: string, sessionStorage: any): TWhatsAppContent[] => {
+
+	// create list for whatsapp content
+	let whatsAppContents: TWhatsAppContent[] = [];
+
+	// Loop through all provided Cogngiy bot messages
+	if (processedOutput.outputStack) {
+		for (let stackItem of processedOutput.outputStack) {
+
+			// check if default text was sent
+			if (stackItem.text && !(stackItem.data && stackItem.data?._cognigy && (stackItem.data?._cognigy?._default))) {
+
+				// send default text
+				whatsAppContents.push({
+					from: sessionId,
+					contentType: "text",
+					text: stackItem.text
+				});
+			}
+
+			// check for location message
+			else if (stackItem.data.location) {
+
+				const { longitude, latitude, name, address } = stackItem.data.location;
+
+				whatsAppContents.push({
+					from: sessionId,
+					contentType: "location",
+					location: {
+						longitude,
+						latitude,
+						name,
+						address
+					},
+				});
+			}
+
+			// check if default templates are defined
+			else if (stackItem.data && (stackItem.data?._cognigy?._default)) {
+
+				let defaultContent = stackItem.data._cognigy._default;
+
+				// look for gallery
+				if (defaultContent._gallery != null && defaultContent._gallery?.items) {
```

Comment:
can be boiled down to (defaultContent._gallery?.items)

Reply:
done